### ISSUE TITLE
Put non-derivative parts of Blender under the MIT or CC license

### ISSUE THREAD

**Issue by capnm** [godotengine/godot-blender-exporter#87]
IMO the test files and the .escn writing part doesn't need to be GPL licensed.

MIT/X11 is a GPL compatible license:
http://www.gnu.org/licenses/license-list.en.html#X11License

which means that MIT-licensed code can still be included with a GPL-licensed  blender add-on, see e.g.:
https://git.blender.org/gitweb/gitweb.cgi/blender-addons.git/blob/HEAD:/io_import_dxf/dxfgrabber/__init__.py

**Comment by sdfgeoff** [comment]
Uh, not that I can see anyone wanting to use our test files for anything at all ever. So to me it doesn't really make a difference if it's GPL or MIT for the test tiles.

I can hypothetically see why teh .escn writing part may be of interest to other people (for those people, it's contained entirely inside `io_scene_godot/structures.py`). However, the file format itself is so simple that it is something like 400 lines long including some handy make-life-easier stuff. The majority of the exporter is handling the conversion from blender data to godot data, and as it depends on the blender data to do so, it cannot be separated from the GPL.
If someone is planning to use python code to make an ESCN exporter, the hard part is not the file format.

There are also a few references to blender code from inside this final stage, but they could be removed easy enough (or separated), but again, it would only allos some 1/20th of the code and a fraction of the total complexity to be non-GPL.

If we do want to change it, now is the time while there are only seven contributors to the repository (and hence only seven people to contact to ensure they are happy with the license changing). However, in my mind, the effort to change the license isn't worth the effect.

The only thing that changing would permit is the use of more liberally licensed blend files in the test scenes, but as these scenes are intended to be very simple, in most cases they will be custom made by the developer for a specific case rather than downloaded from an external source. The only reason it's important at the moment is because we are using a material test sphere (under CC Attribution), and the use of a CC0 test image - both of which are in PR #56 .  While I believe we could comply with these licenses and the GPL, it may be easier to either remake these assets or perhaps relicense the test blend files (which are all made by myself or @Jason0214 -so changing the license of that part of the project probably won't be too hard).

In the end thought, I will ask @reduz to make a call on if we should change the license or not.

**Comment by capnm** [comment]
I agree the difference GPL or MIT isn't essential, MIT would be just nice to have :-) 

> If someone is planning to use python code to make an ESCN exporter, the hard part is not the file format.

Yes, it's a simple INI format, but the fine (undocumented) format details are spread over the entire Godot code. It's nice to have something in one place. IIRC @ndee85 (coa_tools) already asked reduz if the ESCN file writing code could be more separated ;-)

**Comment by capnm** [comment]
You could also take a look at the glTF project.

» the shared code base is organised in Blender dependent and independent packages «
 https://github.com/KhronosGroup/glTF-Blender-IO

**Comment by Raj2032** [comment]
GPL does suck so yeah I would prefer MIT licenses instead.

**Comment by dodgyville** [comment]
GPL is awesome
